taskcluster/scripts/toolchain/android-gradle-dependencies.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+set -ex
+
+function get_abs_path {
+    local file_path="$1"
+    echo "$( cd "$(dirname "$file_path")" >/dev/null 2>&1 ; pwd -P )"
+}
+
+CURRENT_DIR="$(get_abs_path $0)"
+PROJECT_DIR="$(get_abs_path $CURRENT_DIR/../../../..)"
+
+pushd $PROJECT_DIR
+
+. taskcluster/scripts/toolchain/android-gradle-dependencies/before.sh
+
+COMPONENT_REGEX='^  ([-a-z]+):$'
```

Comment:
The code here has definitely outstripped what is reasonable to do in shell, and this should be rewritten in python (parsing yaml is definitely a point at which python should be used).

Reply:
I definitely agree. I changed this logic many times in the PR, and I've thought several times it would just be better in Python.

I'd like to tackle this in a separate bug so I can make all repos use the same python script.